src/topology/algebra/order/lower.lean

```diff
@@ -0,0 +1,271 @@
+/-
+Copyright (c) 2022 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+
+import topology.basic
+import topology.order
+import topology.separation
+import data.set.intervals.basic
+import order.upper_lower
+
+/-!
+# Lower topology
+
+This file introduces the lower topology on a preorder. It is shown that the lower topology on a
+partial order is T₀ and the non-empty complements of the upper closures of finite subsets form a
+basis.
+
+## Implementation notes
+
+Approach inspired by `order_topology` from topology.algebra.order.basic
+
+## References
+
+* [Gierz et al, A Compendium of Continuous Lattices][GierzEtAl1980]
+
+## Tags
+
+lower topology, preorder
+-/
+
+universe u
+variable {α : Type u}
+
+open set topological_space
+
+/--
+Type synonym for a preorder equipped with the lower topology
+-/
+@[reducible]
+def lower (α : Type u) := α
+
+-- Needed if the above definition is not reducible
+/-
+instance lower.preorder (α : Type u) [preorder α] :
+  preorder (lower α) := _inst_1
+-/
+
+/--
+The lower topology is the topology generated by the complements of the closed intervals to infinity
+-/
+def lower_topology (α : Type u) [preorder α] : topological_space α := generate_from {s | ∃ a, (Ici a)ᶜ = s}
+
+instance lower.topological_space (α : Type u) [preorder α] :
+  topological_space (lower α) := lower_topology α
+
+
+section pre_order
+
+
+variable [preorder α]
+
+lemma lower_topology_def : (lower_topology α) = generate_from {s | ∃ a, (Ici a)ᶜ = s} := rfl
+
+lemma is_open_iff_generate_Ici_comp {s : set α} :
+  is_open s ↔ generate_open {s | ∃a, (Ici a)ᶜ = s } s := iff.rfl
+
+/-
+Left-closed right-infinite intervals [a,∞) are closed in the lower topology
+-/
+lemma ici_is_closed (a : α) :
```

Comment:
```suggestion
lemma is_closed_Ici (a : α) :
```
to match `is_closed_Ici`. Maybe there's a way to unify those two lemmas?

Reply:
Not sure - both `is_closed_...` and `..._is_closed` appear elsewhere in mathlib. Which lemma did you think this should be unified with?

Reply:
I think the former is more common and we already have [is_closed_Iic](https://leanprover-community.github.io/mathlib_docs/topology/algebra/order/basic.html#is_closed_Iic) etc. for intervals.

Reply:
Presumably `upper_closure_is_closed` should be `is_closed_upper_closure` and `lower_open_is_lower` should be `is_lower_lower_open` and `lower_basis_is_basis` should be `is_basis_lower_basis`?

Reply:
`is_closed_upper_closure` is how I would name it. I would put everything in the `lower_topology` namespace though, so the other two could be named `is_lower_set_of_is_open` and simply `is_topological_basis`.

Reply:
It's funny how you have basically the same lemmas as in #16975 but under different assumptions. I wonder whether there's a common generalisation...